app/frontend/components/pages/delegations/delegatePage.tsx

```diff
@@ -82,45 +112,65 @@ const BigDelegatorOffer = () => (
 )
 
 interface Props {
-  updateStakePoolIdentifier: any
-  removeStakePool: any
-  stakePool: any
-  delegationFee: any
-  calculatingDelegationFee: any
+  stakePool: Stakepool
+  delegationFee: Lovelace
+  calculatingDelegationFee: boolean
   delegationValidationError: any
-  isShelleyCompatible: any
-  confirmTransaction: any
-  selectAdaliteStakepool: any
-  shouldShowConfirmTransactionDialog: any
-  txSuccessTab: any
+  isShelleyCompatible: boolean
+  shouldShowConfirmTransactionDialog: boolean
+  txSuccessTab: string
   gettingPoolInfo: boolean
-  poolRecommendation: any
-  pool: any
+  poolRecommendation: PoolRecommendation
+  pool: Stakepool
   isBigDelegator: boolean
+  validStakepools: Stakepools
   withAccordion: boolean
   title: string
+  updateStakePoolIdentifier: (poolHash: string) => void
+  confirmTransaction: (txConfirmType) => Promise<void>
+  selectAdaliteStakepool: () => void
 }
 
-class Delegate extends Component<Props, {dropShadow: boolean}> {
-  constructor(props) {
+interface DelegateState {
+  dropShadow: boolean
+  fieldValue: string
+  isTicker: boolean
+  isPoolHash: boolean
+}
+
+class Delegate extends Component<Props, DelegateState> {
```

Comment:
This might be a good chance to refactor this to functional component.

Reply:
Refactored